include/deal.II/lac/vector.templates.h

```diff
@@ -1010,7 +1011,7 @@ template <typename Number>
 void
 Vector<Number>::deallocate()
 {
-  free(values);
+  free(values.get());
```

Comment:
Doesn't this (and the line below) call `free` twice? I guess `values.reset()` should be sufficient here.

Reply:
Uh, oh. You are right. Let me go back to this and fix up a couple of other places where I can simplify things (though they are not wrong).